src/ui/public/persisted_log/persisted_log.js

```diff
@@ -1,42 +1,48 @@
 import { uiModules } from 'ui/modules';
 import _ from 'lodash';
-
-uiModules.get('kibana/persisted_log')
-  .factory('PersistedLog', function ($window, localStorage) {
-    function PersistedLog(name, options) {
-      options = options || {};
-      this.name = name;
-      this.maxLength = parseInt(options.maxLength, 10);
-      this.filterDuplicates = options.filterDuplicates || false;
-      this.items = localStorage.get(this.name) || [];
-      if (!isNaN(this.maxLength)) this.items = _.take(this.items, this.maxLength);
+import { Storage } from 'ui/storage';
+
+const localStorage = new Storage(window.localStorage);
+
+export class PersistedLog {
+  constructor(name, options, storage) {
```

Comment:
We can specify defaults like this to be more efficient:

```js
constructor(name, options = {}, storage = localStorage) {
```

A bit outside the scope of the PR but not that big a change.